apache/superset
# fix(menu): Styling active menu in SPA navigation

### SUMMARY
Fixes the addressed issue (no. 2) on https://github.com/apache/superset/pull/25151#issuecomment-1747279563

This commit replaces Link with NavLink to fix outdated highlighted menu. This commit replaces the ant-menu-item-selected style by NavLink active style to sync the router active link status.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:

https://github.com/apache/superset/assets/1392866/5a4e20a0-e393-4d63-98ad-7789a44a4cce

After:

https://github.com/apache/superset/assets/1392866/23cdd7ea-aed0-44fa-94d2-33dc08c0a03f


### TESTING INSTRUCTIONS
Click Chart (or Dashboard) link
Click Query History menu under the SQL Link
Navigate back by the browser button
See the highlighted menu on the header

### ADDITIONAL INFORMATION
